shared/constants/chat.js

```diff
@@ -649,8 +669,37 @@ const getSelectedRouteState = (state: TypedState) => {
   return getPathState(state.routeTree.routeState, [chatTab, selected])
 }
 
-function messageKey (kind: MessageKeyKind, value: string | number): MessageKey {
-  return `${kind}:${value}`
+function messageKey (conversationIDKey: ConversationIDKey, kind: MessageKeyKind, value: string | number): MessageKey {
+  return `${conversationIDKey}:${kind}:${value}`
+}
+
+function messageKeyValue (key: MessageKey): string {
+  return key.split(':')[2]
+}
+
+function messageKeyConversationIDKey (key: MessageKey): ConversationIDKey {
+  return key.split(':')[0]
+}
+
+function messageKeyKind (key: MessageKey): MessageKeyKind {
+  const [_, kind] = key.split(':') // eslint-disable-line no-unused-vars
```

Comment:
i had some issue doing that in the past. maybe lint or flow or something. if it works now 👍